core/src/main/java/org/apache/calcite/sql/dialect/SparkSqlDialect.java

```diff
@@ -982,4 +986,28 @@ private String removeDotFromAMAndPM(SqlNode dateStringSqlNode) {
     }
     return dateString;
   }
+
+  private void unparseCoalesce(SqlWriter writer, SqlCall call) {
+    final SqlWriter.Frame coalesceFrame = writer.startFunCall("COALESCE");
+    for (SqlNode operand : call.getOperandList()) {
+      writer.sep(",");
+      if (isFormatSqlBasicCall(operand)) {
+        unparseFormatInCoalesce(writer, operand);
+      } else {
+        operand.unparse(writer, 0, 0);
+      }
+    }
+    writer.endFunCall(coalesceFrame);
+  }
+
+  private void unparseFormatInCoalesce(SqlWriter writer, SqlNode call) {
+    final SqlWriter.Frame stringFrame = writer.startFunCall("STRING");
+    ((SqlCall) call).operand(1).unparse(writer, 0, 0);
+    writer.endFunCall(stringFrame);
+  }
+
+  private boolean isFormatSqlBasicCall(SqlNode sqlNode) {
```

Comment:
Plz move isFormatSqlBasicCall to Util method and use SqlKind.FORMAT.name() instead "FORMAT" in condition.

Reply:
Done